### ISSUE TITLE
[RMP] T4R fixes: MultiGPU data parallel training for next-item prediction and fixed serving

### ISSUE THREAD

**Issue by karlhigley** [NVIDIA-Merlin/Merlin#522]
## Problem:
We have customers who would like to use multi-GPU Transformers4Rec but are blocked by issues with our existing support for session-based models.

## Goal:
- Unblock customer use cases so they can try out T4R to give us feedback

## Constraints:
- We don't yet have Torchscript support (which is out of scope this issue)

## Starting Point:

- [x] Enable `DataParallel` / `DistributedDataParallel` training using HF Trainer for next-item prediction
  - [x] Next item prediction - https://github.com/NVIDIA-Merlin/Transformers4Rec/issues/473 - `DataParallel` works if the model is wrapped manually by the user (i.e. `model = torch.nn.DataParallel(model)` for training, but that wrapping should happen automatically by the HF Trainer [here](https://github.com/huggingface/transformers/blob/d6eeb871706db0d64ab9ffd79f9545d95286b536/src/transformers/trainer.py#L1327)
  - [x] https://github.com/NVIDIA-Merlin/Transformers4Rec/pull/483
  - [x] https://github.com/NVIDIA-Merlin/Transformers4Rec/pull/496#pullrequestreview-1131707972
  - [x] NVIDIA-Merlin/Transformers4Rec#492

- [x] Fix the serving sections of the existing T4R notebooks
  - [x] https://github.com/NVIDIA-Merlin/NVTabular/pull/1628
  - [x] https://github.com/NVIDIA-Merlin/Transformers4Rec/pull/468 
  
- [x] [Task] Add multi-GPU example for Transformer4Rec PyTorch (https://github.com/NVIDIA-Merlin/Transformers4Rec/issues/508)
- [ ] https://github.com/NVIDIA-Merlin/Transformers4Rec/issues/526

Note: The multi-GPU training of the specific use cases of session binary classification / regression is addressed by [RMP #708](https://github.com/NVIDIA-Merlin/Merlin/issues/708)

**Comment by karlhigley** [comment]
Distributed Data Parallel training is something we want to do, but I don't think it's part of this effort to fix the immediate blockers. Does that match what y'all understand, @EvenOldridge @viswa-nvidia?

**Comment by sohn21c** [comment]
Binary Classification DP training would unblock the issue although DDP is preferred if more performant. More details are captured [here](https://github.com/NVIDIA-Merlin/Transformers4Rec/issues/423)

**Comment by karlhigley** [comment]
Either DistributedDataParallel training is part of the scope of the quick fixes or it isn't, and it sounds like it isn't so we should track that work somewhere (but not here.)

**Comment by viswa-nvidia** [comment]
@gabrielspmoreira , please create a ticket for DP training binary classification and link it here.

**Comment by gabrielspmoreira** [comment]
@viswa-nvidia @EvenOldridge 
Me, @rnyak, @sararb and @nzarif met today about the issues related to `DataParallel`.
We tested `DataParallel` for Next Item Prediction for one of the examples and it is not working, differently from what Sara found some weeks ago in another example.
So we have both Next Item Prediction and Binary Classification not working with DataParallel currently.
We have associated the issues for both in this RMP ticket description.
Should we remove the scope of DataParallel from this RMP and create another RMP ticket focused in DataParallel support (targeted for release 22.09)?
